server/service.go

```diff
@@ -431,7 +445,33 @@ func (m *BoostService) handleGetPayload(w http.ResponseWriter, req *http.Request
 	requestCtx, requestCtxCancel := context.WithCancel(context.Background())
 	defer requestCtxCancel()
 
-	for _, relay := range m.relays {
+	key := bidRespKey{
+		slot:      payload.Message.Slot,
+		blockHash: payload.Message.Body.ExecutionPayloadHeader.BlockHash.String(),
+	}
+
+	// It means that someone is trying to call getPayload without having called getHeader first.
+	fmt.Printf("KEY %+v\n", key)
+	fmt.Printf("BIDS %+v\n", m.bids[key])
+	if m.bids[key] == nil {
+		// TODO: Log error.
+		// log.WithField("", "").Errorf("")
+		m.respondError(w, http.StatusBadGateway, errNoSuccessfulRelayResponse.Error())
+		return
+	}
+	publicKeys := m.bids[key].proposerPublicKeys
+	publicKey := types.PublicKey{}
+
+	for _, publicKey = range publicKeys {
+		ok, _ := types.VerifySignature(payload.Message, m.builderSigningDomain, publicKey[:], payload.Signature[:])
```

Comment:
that signature check doesn't fail on invalid signature. i think we can leave that to the relay too, we don't check the signatures in other places in mev-boost either.

Reply:
The issue is that I need this check to get the public key associated to the current proposer.

1. I want to get the current proposer's public key in order to call its relays (based on the configuration file).
2. Because this value is not given as a parameter in the `getPayload` call, I use the list of public keys made from the various calls to `getHeader` associated to this `slot + blockchash` (bidKey). One of these public keys is the proposer's one, the only way to know which one it is is to try to verify the signature of the `getPayload` parameter : if it indeed verifies the signature, it's the public key I'm looking for. Otherwise it's an another proposer's public key.

Reply:
Alright, removed in eb6cbc1543f2f572b298ec6c663f9c34263d7793.